test/unit/plotly-dash-preview_test.js

```diff
@@ -0,0 +1,72 @@
+const tap = require('tap')
+const sinon = require('sinon')
+
+const _module = require('../../src/component/plotly-dash-preview')
+const remote = require('../../src/util/remote')
+const { createMockWindow } = require('../common')
+
+tap.test('parse:', t => {
+  const fn = _module.parse
+
+  t.test('should error when no *url* field is given', t => {
+    const shouldFail = ['', null, true, 1, {}]
+
+    shouldFail.forEach(d => {
+      t.test(`(case ${JSON.stringify(d)})`, t => {
+        fn({url: d}, {}, (errorCode, result) => {
+          t.equal(errorCode, 400)
+          t.end()
+        })
+      })
+    })
+
+    t.end()
+  })
+
+  t.end()
+})
+
+tap.test('render:', t => {
+  const fn = _module.render
+
+  t.afterEach((done) => {
+    remote.createBrowserWindow.restore()
+    done()
+  })
+
+  t.test('should call printToPDF', t => {
+    const win = createMockWindow()
+    sinon.stub(remote, 'createBrowserWindow').returns(win)
+    win.webContents.executeJavaScript.resolves(true)
+    win.webContents.printToPDF.yields(null, '-> image data <-')
+
+    fn({
+      url: 'dummy'
+    }, {}, (errorCode, result) => {
+      t.ok(win.webContents.printToPDF.calledOnce)
+      t.ok(win.close.calledOnce)
+      t.equal(errorCode, undefined, 'code')
+      t.equal(result.imgData, '-> image data <-', 'result')
+      t.end()
+    })
+  })
+
+  t.test('should handle printToPDF errors', t => {
+    const win = createMockWindow()
+    sinon.stub(remote, 'createBrowserWindow').returns(win)
+    win.webContents.executeJavaScript.resolves(true)
```

Comment:
Nice tests! Like mentioned in an earlier comment, we'll need to handle (and test :smirk: ) the case where `executeJavascript` does not resolve.